Sources/PersistentCollections/Node/_Node+isEqual.swift

```diff
@@ -12,9 +12,9 @@
 // TODO: `Equatable` needs more test coverage, apart from hash-collision smoke test
 extension _Node {
   @inlinable
-  internal func isEqual(
-    to other: _Node,
-    by areEquivalent: (Element, Element) -> Bool
+  internal func isEqual<Value2>(
+    to other: _Node<Key, Value2>,
+    by areEquivalent: (Value, Value2) -> Bool
   ) -> Bool {
```

Comment:
Why is it necessary to introduce the additional generic `Value2` here?

Reply:
This isn't exercised, and there is a chance it won't be, in which case we can remove it later. (But it doesn't really have a performance drawback, so it's no big deal if it stays in.)

The reason I wanted to leave this door open is in case we want to tackle a weird asymmetry in SetAlgebra. It is customary for SetAlgebra types to provide flexible overloads for `isSubset`/`isDisjoint`/`intersection`/etc that can take more types than just `Self` -- elsewhere I usually have an unconstrained generic overload that can take any `Sequence` plus a handful of concrete overloads for select types for which I can do a more efficient implementation. (So e.g. `OrderedSet` provides not only `isSubset(of other: Self)`, but also `isSubset(of other: Set<Element>)` and `isSubset(of other: some Sequence<Element>)`.) The asymmetry I find weird is that this applies to all such operations except `Equatable.==` -- why can't I easily check if a `Set` and an `OrderedSet` contain exactly the same members?

In the case of `PersistentSet`, we currently have `isSubset(of:)` defined for `Self` and `PersistentDictionary<Element, T>.Keys` arguments. (Later on we will also need one that can take any `SetAlgebra` and one that can take any `Sequence`.) If we do end up adding an `isEqual(to:)` family of methods that generalize `==`, then we will need to this particular flavor of generic definition in `_Node` to be able to compare a persistent set to the `Keys` view of any persistent dictionary that has a matching `Key` type.